### ISSUE TITLE
https urls cannot be fetched using socks

### ISSUE THREAD

**Issue by volth** [frugalmechanic/fm-http#18]
| | ```http``` | ```https``` |
| -- | -- | -- |
| without socks | ok | ok |
| with socks | ok | FAIL |

testcase:

```scala
import scala.concurrent.duration._
import scala.concurrent.{Future, Await}
import scala.concurrent.ExecutionContext.Implicits.global
import `com.frugalmechanic::fm-http:0.17.0`
import fm.http.client.{FullStringResponse, HttpClient, Response}

object FmHttpTest extends App {
  val client = HttpClient(socksProxy = Some("127.0.0.1" -> 9063)) // Tor socks
  val res: Future[FullStringResponse] = client.getFullString("https://bgp.he.net/")
  println(Await.result(res, 10.minute).body)
}
```

stacktrace:
```
19:56:10.127 [main] DEBUG io.netty.util.internal.logging.InternalLoggerFactory - Using SLF4J as the default logging framework
19:56:10.184 [main] DEBUG io.netty.util.internal.PlatformDependent0 - -Dio.netty.noUnsafe: false
19:56:10.191 [main] DEBUG io.netty.util.internal.PlatformDependent0 - Java version: 8
19:56:10.194 [main] DEBUG io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: available
19:56:10.195 [main] DEBUG io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.copyMemory: available
19:56:10.196 [main] DEBUG io.netty.util.internal.PlatformDependent0 - java.nio.Buffer.address: available
19:56:10.199 [main] DEBUG io.netty.util.internal.PlatformDependent0 - direct buffer constructor: available
19:56:10.200 [main] DEBUG io.netty.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: available, true
19:56:10.200 [main] DEBUG io.netty.util.internal.PlatformDependent0 - jdk.internal.misc.Unsafe.allocateUninitializedArray(int): unavailable prior to Java9
19:56:10.201 [main] DEBUG io.netty.util.internal.PlatformDependent0 - java.nio.DirectByteBuffer.<init>(long, int): available
19:56:10.201 [main] DEBUG io.netty.util.internal.PlatformDependent - sun.misc.Unsafe: available
19:56:10.201 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.tmpdir: /home/user/tmp (java.io.tmpdir)
19:56:10.202 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.bitMode: 64 (sun.arch.data.model)
19:56:10.203 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.noPreferDirect: false
19:56:10.204 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.maxDirectMemory: 3715629056 bytes
19:56:10.204 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.uninitializedArrayAllocationThreshold: -1
19:56:10.205 [main] DEBUG io.netty.util.internal.CleanerJava6 - java.nio.ByteBuffer.cleaner(): available
19:56:10.244 [main] DEBUG io.netty.channel.MultithreadEventLoopGroup - -Dio.netty.eventLoopThreads: 8
19:56:10.269 [main] DEBUG io.netty.channel.nio.NioEventLoop - -Dio.netty.noKeySetOptimization: false
19:56:10.269 [main] DEBUG io.netty.channel.nio.NioEventLoop - -Dio.netty.selectorAutoRebuildThreshold: 512
19:56:10.290 [main] DEBUG io.netty.util.internal.PlatformDependent - org.jctools-core.MpscChunkedArrayQueue: available
19:56:10.397 [main] DEBUG io.netty.handler.ssl.OpenSsl - netty-tcnative not in the classpath; OpenSslEngine will be unavailable.
19:56:11.192 [main] DEBUG io.netty.handler.ssl.JdkSslContext - Default protocols (JDK): [TLSv1.2, TLSv1.1, TLSv1]
19:56:11.192 [main] DEBUG io.netty.handler.ssl.JdkSslContext - Default cipher suites (JDK): [TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_
WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_CBC_SHA]
19:56:11.278 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.level: simple
19:56:11.278 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.targetRecords: 4
19:56:11.296 [main] INFO fm.http.client.DefaultHttpClient - GET https://bgp.he.net/
19:56:11.393 [main] DEBUG io.netty.util.internal.InternalThreadLocalMap - -Dio.netty.threadLocalMap.stringBuilder.initialSize: 1024
19:56:11.393 [main] DEBUG io.netty.util.internal.InternalThreadLocalMap - -Dio.netty.threadLocalMap.stringBuilder.maxSize: 4096
19:56:11.425 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.allocator.type: unpooled
19:56:11.426 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.threadLocalDirectBufferSize: 65536
19:56:11.426 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.maxThreadLocalCharBufferSize: 16384
19:56:11.669 [fm-http-client-worker-1] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.bytebuf.checkAccessible: true
19:56:11.672 [fm-http-client-worker-1] DEBUG io.netty.util.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@2f0bb74b
19:56:11.768 [fm-http-client-worker-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxCapacity.default: 32768
19:56:11.769 [fm-http-client-worker-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxSharedCapacityFactor: 2
19:56:11.769 [fm-http-client-worker-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.linkCapacity: 16
19:56:11.769 [fm-http-client-worker-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.ratio: 8
Exception in thread "main" io.netty.handler.codec.UnsupportedMessageTypeException: io.netty.handler.codec.socks.SocksInitRequest (expected: io.netty.buffer.ByteBuf)
        at io.netty.handler.ssl.SslHandler.write(SslHandler.java:675)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:732)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:724)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:809)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:717)
        at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.write(CombinedChannelDuplexHandler.java:528)
        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:101)
        at io.netty.channel.CombinedChannelDuplexHandler.write(CombinedChannelDuplexHandler.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:732)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:795)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:807)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:788)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:824)
        at fm.http.client.NettyHttpClientPipelineHandler.writeSocksInit(NettyHttpClientPipelineHandler.scala:224)
        at fm.http.client.NettyHttpClientPipelineHandler.write(NettyHttpClientPipelineHandler.scala:347)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:732)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:795)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:807)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:788)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:824)
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1025)
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:259)
        at fm.http.client.DefaultHttpClient.$anonfun$makeNewChannel$2(DefaultHttpClient.scala:344)
        at fm.netty.NettyFutureWrapper$Listener.operationComplete(NettyFutureWrapper.scala:27)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:500)
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:479)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:420)
        at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
        at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:82)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:257)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:292)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:634)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:581)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:460)
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:131)
        at java.lang.Thread.run(Thread.java:748)
19:56:12.090 [scala-execution-context-global-22] WARN fm.http.client.DefaultHttpClient - Request Failed: GET https://bgp.he.net/
io.netty.handler.codec.UnsupportedMessageTypeException: io.netty.handler.codec.socks.SocksInitRequest (expected: io.netty.buffer.ByteBuf)
        at io.netty.handler.ssl.SslHandler.write(SslHandler.java:675)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:732)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:724)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:809)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:717)
        at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.write(CombinedChannelDuplexHandler.java:528)
        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:101)
        at io.netty.channel.CombinedChannelDuplexHandler.write(CombinedChannelDuplexHandler.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:732)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:795)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:807)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:788)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:824)
        at fm.http.client.NettyHttpClientPipelineHandler.writeSocksInit(NettyHttpClientPipelineHandler.scala:224)
        at fm.http.client.NettyHttpClientPipelineHandler.write(NettyHttpClientPipelineHandler.scala:347)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:732)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:795)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:807)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:788)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:824)
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1025)
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:259)
        at fm.http.client.DefaultHttpClient.$anonfun$makeNewChannel$2(DefaultHttpClient.scala:344)
        at fm.netty.NettyFutureWrapper$Listener.operationComplete(NettyFutureWrapper.scala:27)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:500)
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:479)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:420)
        at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
        at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:82)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:257)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:292)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:634)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:581)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:460)
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:131)
        at java.lang.Thread.run(Thread.java:748)
19:56:12.096 [TaskRunner Shutdown Hook - HttpExecutionContext.Timer] WARN fm.common.ScheduledTaskRunner - Shutting down HttpExecutionContext.Timer with 1 items still in queue!
```

**Comment by er1c** [comment]
Looks related to: https://github.com/frugalmechanic/fm-http/blob/11a583ae9452743cc4df17e5937ef3aea2154331/src/main/scala/fm/http/client/DefaultHttpClient.scala#L310-L314

It looks like Netty `4.1` has built in `Socks4ProxyHandler` (https://netty.io/4.1/api/io/netty/handler/proxy/Socks4ProxyHandler.html) that looks interesting

**Comment by volth** [comment]
No problem as with ```fm-http 0.23```

**Comment by tpunder** [comment]
Yes, this was fixed in `0.23.0` via https://github.com/frugalmechanic/fm-http/commit/bd505565aabdd1a698be383a724dcd7222079410.  I now use the Netty 4.1 built-in SOCKS proxy support.
